lib/beaker-pe/pe-client-tools/install_helper.rb

```diff
@@ -0,0 +1,109 @@
+              package_name = product.dup
+              package_name << "-#{opts[:pe_client_tools_version]}-1.#{variant}#{version}.#{arch}.rpm" if opts[:pe_client_tools_version]
+            when /debian|ubuntu|cumulus|huaweios/
+              release_path = "#{opts[:dev_builds_url]}/#{product}/#{ opts[:pe_client_tools_sha] }/artifacts/deb/#{codename}/#{opts[:puppet_collection]}"
+              package_name = product.dup
+              package_name << "_#{opts[:pe_client_tools_version]}-1#{host['platform'].codename}_#{arch}.deb" if opts[:pe_client_tools_version]
+            when /windows/
+              release_path = "#{opts[:dev_builds_url]}/#{product}/#{ opts[:pe_client_tools_sha] }/artifacts/#{variant}/#{opts[:puppet_collection]}/x#{arch}"
+              package_name = product.dup
+              package_name << "-#{opts[:pe_client_tools_version]}.1-x#{arch}_VANAGON.msi" if opts[:pe_client_tools_version]
+            when /osx/
+              release_path = "#{opts[:dev_builds_url]}/#{product}/#{ opts[:pe_client_tools_sha] }/artifacts/apple/#{version}/#{opts[:puppet_collection]}/#{arch}"
+              package_base = product.dup
+              package_base << "-#{opts[:pe_client_tools_version]}" if opts[:pe_client_tools_version]
+              package_name = package_base.dup
+              package_name << '-1' if opts[:pe_client_tools_version]
+              installer    = package_name + '-installer.pkg'
+              package_name << ".#{variant}#{version}.dmg" if opts[:pe_client_tools_version]
+            else
+              raise "install_puppet_agent_on() called for unsupported " +
+                "platform '#{host['platform']}' on '#{host.name}'"
+            end
+
+            if package_name
+              case host['platform']
+              when /windows/
+                install_msi_on(host, File.join(release_path, package_name), {}, opts)
+              else
+                copy_dir_local = File.join('tmp', 'repo_configs')
+                fetch_http_file(release_path, package_name, copy_dir_local)
+                scp_to host, File.join(copy_dir_local, package_name), host.external_copy_base
+
+                if host['platform'] =~ /debian|ubuntu|cumulus|huaweios/
+                  on host, "dpkg -i #{package_name}"
+                elsif host['platform'] =~ /osx/
+                  install_dmg_on(host, package_name, package_base, installer, opts)
+                else
+                  host.install_package( package_name )
+                end
+              end
+            end
+          end
+          #}}}
+        end
+
+        def install_dmg_on(host, dmg_path, pkg_base, pkg_name, opts = {})
+          dmg_name = File.basename(dmg_path, '.dmg')
+          on(host, "hdiutil attach #{dmg_name}.dmg")
+          on(host, "installer -pkg /Volumes/#{pkg_base}/#{pkg_name} -target /")
+        end
+
+        def install_msi_on(hosts, msi_path, msi_opts = {}, opts = {})
+          #{{{
```

Comment:
Thanks for looking into this @johnduarte! Yeah, the `windows_utils.rb` file is a problem, I think most of those methods should be put into host objects. But for now, that's not a yak you need to shave.

I would recommend doing the refactor for the mac method, and ticketing the windows work with the information you have now, so that hopefully we'll be able to get around to this at some point soon.